projects/super_res/super_res.py

```diff
@@ -0,0 +1,203 @@
+import xarray as xr
+import numpy as np
+import fv3viz
+from vcm.catalog import catalog
+import matplotlib.pyplot as plt
+import os
+import cartopy.crs as ccrs
+
+GRID = catalog["grid/c384"].read()
+
+DPI = 800
+
+
+def repeat_arr(arr, upscale_factor):
+    # This repeats values along the last dimension
+    return np.repeat(arr, upscale_factor, axis=-1)
+
+
+def upscale_nearest(ds, upscale_factor, dim):
+    ds_upscaled = xr.apply_ufunc(
+        repeat_arr,
+        ds,
+        input_core_dims=[[dim]],  # apply function along this dimension
+        output_core_dims=[[dim]],  # the output will also have this dimension
+        kwargs={
+            "upscale_factor": upscale_factor
+        },  # pass in other arguments to the function
+        dask="parallelized",  # needed for dask arrays
+        output_sizes={
+            dim: ds.sizes[dim] * upscale_factor
+        },  # the size of the output dimension
+        exclude_dims=set((dim,)),  # allow changing the dimension size
+    )
+    return ds_upscaled
+
+
+def plot_pattern_bias(output: xr.Dataset, target: xr.Dataset, varname: str):
+    """Plot the pattern bias between two datasets
+
+    Args:
+        output (xr.Dataset): output dataset
+        target (xr.Dataset): target dataset
+    """
+    # compute the pattern bias
```

Comment:
nit: Might as well make this a separate function?

Reply:
Maybe, but I think this is too much on the elegance side of elegant vs explicit for a single-user plotting script. When I make plotting routines I find it helpful to be able to change how and what is plotted when editing the same function, and I generally keep both those pieces of information in my working context/memory.